### ISSUE TITLE
Network connection not working

### ISSUE THREAD

**Issue by wdolek** [docker/for-win#755]
### Expected behavior
Connection to work.

### Actual behavior
The remote name could not be resolved: 'downloads.mongodb.org'

### Information

Running on notebook with more network interfaces (wifi, eth). Tried reinstall/factory reset, still not able to make container connect to any remote host.

```
> docker version
Client:
 Version:      17.03.1-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Tue Mar 28 00:40:02 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.03.1-ce
 API version:  1.27 (minimum version 1.24)
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Tue Mar 28 00:40:02 2017
 OS/Arch:      windows/amd64
 Experimental: false
```

```
> docker info
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 5
Server Version: 17.03.1-ce
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: l2bridge l2tunnel nat null overlay transparent
Swarm: inactive
Default Isolation: hyperv
Kernel Version: 10.0 15063 (15063.0.amd64fre.rs2_release.170317-1834)
Operating System: Windows 10 Pro
OSType: windows
Architecture: x86_64
CPUs: 4
Total Memory: 31.89 GiB
Name: *
ID: *
Docker Root Dir: C:\ProgramData\Docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: -1
 Goroutines: 16
 System Time: 2017-05-17T10:47:49.853679+02:00
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

```
> docker network
NETWORK ID          NAME                DRIVER              SCOPE
3c8dc74e254f        nat                 nat                 local
7977ba28a759        none                null                local
```

```
Downloading http://downloads.mongodb.org/win32/mongodb-win32-x86_64-2008plus-ssl-3.4.4-signed.msi ...
Exception calling "DownloadFile" with "2" argument(s): "The remote name could
not be resolved: 'downloads.mongodb.org'"
At line:1 char:97
+ ... NLOAD_URL); (New-Object System.Net.WebClient).DownloadFile($env:MONGO ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], ParentContainsErrorRecordE
   xception
    + FullyQualifiedErrorId : WebException

The command 'powershell -Command $ErrorActionPreference = 'Stop'; Write-Host ('Downloading {0} ...' -f $env:MONGO_DOWNLOAD_URL);     (New-Object System.Net.WebClient).DownloadFile($env:MONGO_DOWNLOAD_URL, 'mongo.msi');         Write-Host ('Verifying sha256 ({0}) ...' -f $env:MONGO_DOWNLOAD_SHA256);     if ((Get-FileHash mongo.msi -Algorithm sha256).Hash -ne $env:MONGO_DOWNLOAD_SHA256) {         Write-Host 'FAILED!';         exit 1;     };         Write-Host 'Installing ...';     Start-Process msiexec -Wait         -ArgumentList @(             '/i',             'mongo.msi',             '/quiet',             '/qn',             'INSTALLLOCATION=C:\mongodb',             'ADDLOCAL=all'         );     $env:PATH = 'C:\mongodb\bin;' + $env:PATH;     [Environment]::SetEnvironmentVariable('PATH', $env:PATH, [EnvironmentVariableTarget]::Machine);         Write-Host 'Verifying install ...';     Write-Host '  mongo --version'; mongo --version;     Write-Host '  mongod --version'; mongod --version;         Write-Host 'Removing ...';     Remove-Item C:\mongodb\bin\*.pdb -Force;     Remove-Item C:\windows\installer\*.msi -Force;     Remove-Item mongo.msi -Force;         Write-Host 'Complete.';' returned a non-zero code: 1
```

(file is accessible)

Tried to eliminate DNS issue with replacing host name by IP (52.222.157.106) with result:

```
Exception calling "DownloadFile" with "2" argument(s): "Unable to connect to the remote server"
```

Tried Linux containers - work.

### Steps to reproduce the behavior

  1. Install Docker for Windows from https://store.docker.com/editions/community/docker-ce-desktop-windows
  2. Switch to `Windows containers`
  3. Download `mongodb` Dockerfile from https://hub.docker.com/_/mongo/
  4. Run `docker build .` in folder with Dockerfile
  5. Wait ... and get error above

**Comment by wdolek** [comment]
I downloaded `Debug-ContainerHost.ps1` and run it and got these errors:

```
Checking for common problems                                                                                                                                                               
Describing Windows Version and Prerequisites                                                                                                                                               
 [+] Is Windows 10 Anniversary Update or Windows Server 2016 552ms                                                                                                                         
 [+] Has KB3192366, KB3194496, or later installed if running Windows build 14393 116ms                                                                                                     
 [+] Is not a build with blocking issues 25ms                                                                                                                                              
 [+] Has 'Containers' feature installed 45.36s                                                                                                                                             
Describing Docker is installed                                                                                                                                                             
 [+] A Docker service is installed - 'Docker' or 'com.Docker.Service'  69ms                                                                                                                
 [+] Service is running 27ms                                                                                                                                                               
 [+] Docker.exe is in path 2.17s                                                                                                                                                           
 [+] Docker is registered in the EventLog service 145ms                                                                                                                                    
Describing User has permissions to use Docker daemon                                                                                                                                       
 [+] docker.exe should not return access denied 44ms                                                                                                                                       
Describing Windows container settings are correct                                                                                                                                          
 [+] Do not have DisableVSmbOplock set to 1 39ms                                                                                                                                           
 [+] Do not have zz values set 33ms                                                                                                                                                        
 [+] Do not have FDVDenyWriteAccess set to 1 26ms                                                                                                                                          
Describing The right container base images are installed                                                                                                                                   
 [+] At least one of 'microsoft/windowsservercore' or 'microsoft/nanoserver' should be installed 100ms                                                                                     
Describing Container network is created                                                                                                                                                    
Get-VMSwitch : Hyper-V was unable to find a virtual switch with name "nat".                                                                                                                
At C:\Users\zdenek\Workspace\repos\github.com\Microsoft\Virtualization-Documentation\windows-server-container-tools\Debug-ContainerHost\Debug-ContainerHost.ps1:165 char:22                
+ ...      $switchType = (Get-VMSwitch -SwitchName $natVMSwitchName).Switch ...                                                                                                            
+                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                        
    + CategoryInfo          : InvalidArgument: (nat:String) [Get-VMSwitch], VirtualizationException                                                                                        
    + FullyQualifiedErrorId : InvalidParameter,Microsoft.HyperV.PowerShell.Commands.GetVMSwitch                                                                                            
                                                                                                                                                                                           
 [+] At least one local container network is available 2.71s                                                                                                                               
 [+] At least one NAT, Transparent, or L2Bridge Network exists 86ms                                                                                                                        
 [-] NAT Network's vSwitch is internal 277ms                                                                                                                                               
   Expected: {Internal}                                                                                                                                                                    
   But was:  {}                                                                                                                                                                            
   211:       $switchType | Should Be "Internal"                                                                                                                                           
   at <ScriptBlock>, C:\Users\zdenek\Workspace\repos\github.com\Microsoft\Virtualization-Documentation\windows-server-container-tools\Debug-ContainerHost\Debug-ContainerHost.ps1: line 211
 [+] A Windows NAT is configured if a Docker NAT network exists 579ms                                                                                                                      
 [-] Specified Network Gateway IP for NAT network is assigned to Host vNIC 70ms                                                                                                            
   Expected: {True}                                                                                                                                                                        
   But was:  {False}                                                                                                                                                                       
   235:       $vmNicGatewayIPExists | Should Be $true                                                                                                                                      
   at <ScriptBlock>, C:\Users\zdenek\Workspace\repos\github.com\Microsoft\Virtualization-Documentation\windows-server-container-tools\Debug-ContainerHost\Debug-ContainerHost.ps1: line 235
 [+] NAT Network's internal prefix does not overlap with external IP' 49ms                                                                                                                 
```

I'm not really sure why it can't find switch `nat` when `docker network` lists that. How do I fix this?

**Comment by wdolek** [comment]
Tried on another machine where everything works out-of-the box. This issue is related to particular Windows configuration (I've noticed container network has DNS of company VPN - even when I'm not connected to it).

Closing this issue.
